### DISCUSSION TITLE
Is there any guide for iNav setting up AltHold & PosHold?

### DISCUSSION THREAD

**Discussion by lida2003** [iNavFlight/inav#8632] [Q&A]
I found that AltHold and PosHold is NOT that easy to achieve. And there is many factors or pre-setting should be considered.

So is there any guide for iNav setting up AltHold & PosHold? something like 101 lessons.

**Comment by MrD-RC** [comment]
Please specify if you're wanting help with fixed wing or a multirotor.

**Reply by lida2003** [reply]
oh, yes. multirotor 
- 2.5inch 
- MATEK optical flow & LIDAR 3901 module
- FC with baro, NO GPS

Plan to use indoor posHold & AltHold. Currently, result is NOT that good according to guide form MATEK.

[Optical FLow & Lidar Sensor 3901-L0X](http://www.mateksys.com/?portfolio=3901-l0x)

[Setting up Optic Flow in INAV](http://www.mateksys.com/downloads/setup_optic%20flow_in_INAV_rev1.pdf)   PDF manual by digitalentity

**Comment by lida2003** [comment]
I have did following tests and result shows is NOT good as https://www.youtube.com/watch?v=YkIFkhktxoA

a) test environment is the same as tello (tello did well when there is no wind), so I don't think  illumination and surface texture is an issue. Here is my video (Really bad): https://www.bilibili.com/video/BV14G4y1u73E/
==> Or there is anyway to adjust matek module? or guide to fine tune the inav configurations?

b) PID for navigation and all other settings is configured as mateck suggesttion: http://www.mateksys.com/downloads/setup_optic%20flow_in_INAV_rev1.pdf

I'll check motor vibration and propeller (when new propeller comes), although it seems fine with the data(I didn't see or I don't know if it OK), may be I have to punch the throttle more to see the vibration. 

![图片](https://user-images.githubusercontent.com/1296171/209250266-2d41b05b-c82e-4feb-8f2d-74169f576ddb.png)


[blackbox_log_2022-12-20_133509.zip](https://github.com/iNavFlight/inav/files/10291258/blackbox_log_2022-12-20_133509.zip)
